### PR TITLE
Update temp sensor code

### PR DESCRIPTION
**Pull request by khanr107**
temp data logging code

### CODE REVIEW

**Comment by ShyavanS** [comment]
Looks good, but where did you get the info for the kalman filter setup you are using? Mind sending over the source, just curious? Will any of the initial parameters you set need to be tuned as we do trial runs? Also make sure whoever is doing the competition codebase version of the temp sensor task & the IMU tasks also is on the same page on what you did for the kalman filtering. We ideally want to keep everything to the same algorithm to keep our filtering consistent across sensors & codebases.

Only issue I see with the code though is you'll have to update the code with the new pin assignments & replace 23 with A1. Don't forget when you do this to pull changes from the main branch to your branch too so they're both up-to-date as I made some changes to the main branch for the pcb.

File: Car_Code/cheme_car_datalogging_codebase/cheme_car_datalogging_codebase.ino
Comment (ShyavanS) [inline Car_Code/cheme_car_datalogging_codebase/cheme_car_datalogging_codebase.ino]: Change this from 23 to the new pin assignments (should be A1). Don't forget when you do this to pull changes from the main branch to your branch too so they're both up-to-date as I made some changes to the main branch for the pcb.

File: Car_Code/cheme_car_datalogging_codebase/cheme_car_datalogging_codebase.ino
Comment (ShyavanS) [inline Car_Code/cheme_car_datalogging_codebase/cheme_car_datalogging_codebase.ino:52]: Where did you get the info for the kalman filter setup you are using? Mind sending over the source, just curious?

File: Car_Code/cheme_car_datalogging_codebase/cheme_car_datalogging_codebase.ino
Comment (ShyavanS) [inline Car_Code/cheme_car_datalogging_codebase/cheme_car_datalogging_codebase.ino:32]: Will any of the initial parameters you set need to be tuned as we do trial runs?

**Review by ShyavanS — CHANGES_REQUESTED** [review]
Looks good for the most part, just have some questions & one thing to resolve. Also, make sure whoever is doing the competition codebase version of the temp sensor task & the IMU tasks also is on the same page on what you did for the kalman filtering. We ideally want to keep everything to the same algorithm to keep our filtering consistent across sensors & codebases.

File: Car_Code/cheme_car_datalogging_codebase/cheme_car_datalogging_codebase.ino
Comment (khanr107) [reply Car_Code/cheme_car_datalogging_codebase/cheme_car_datalogging_codebase.ino:52]: I used this video  (pseudo code at 6:48)
https://www.youtube.com/watch?v=ruB917YmtgE&t=95s&ab_channel=VDEngineering

Comment (ShyavanS) [reply Car_Code/cheme_car_datalogging_codebase/cheme_car_datalogging_codebase.ino:52]: Okay sounds good!

File: Car_Code/cheme_car_datalogging_codebase/cheme_car_datalogging_codebase.ino
Comment (khanr107) [reply Car_Code/cheme_car_datalogging_codebase/cheme_car_datalogging_codebase.ino:32]: Yes, these parameters need to be tuned based on the characteristics of the system and the noise in our measurements. All of these would be determined while doing trial run. like for example q needs to be increased in the case of our system being very dynamic.

**Review by ShyavanS — APPROVED** [review]
Looks good now! Should be good to merge once Preet reads over & approves it.

**Review by batrap5 — APPROVED** [review]

**Pull request merged**
